Userland/Games/FlappyBug/Game.h

```diff
@@ -0,0 +1,152 @@
+        const RefPtr<Gfx::Bitmap> falling_bitmap { Gfx::Bitmap::load_from_file("/res/icons/flappybug/falling.png") };
+        const RefPtr<Gfx::Bitmap> flapping_bitmap { Gfx::Bitmap::load_from_file("/res/icons/flappybug/flapping.png") };
+        float y {};
+        float velocity {};
+
+        void reset()
+        {
+            y = starting_y;
+        }
+
+        RefPtr<Gfx::Bitmap> current_bitmap() const
+        {
+            return velocity < 0 ? falling_bitmap : flapping_bitmap;
+        }
+
+        Gfx::FloatRect rect() const
+        {
+            return { x - radius, y - radius, radius * 2, radius * 2 };
+        }
+
+        void flap()
+        {
+            const float flap_strength = 15.0f;
+            velocity = -flap_strength;
+        }
+
+        void fall()
+        {
+            const float gravity = 1.0f;
+            velocity += gravity;
+        }
+
+        void apply_velocity()
+        {
+            y += velocity;
+        }
+    };
+
+    struct Obstacle {
+        const float width { 20 };
+        Color color { Color::DarkGray };
+        float x;
+        float gap_top_y { 200 };
+        float gap_height { 175 };
+
+        void reset()
+        {
+            x = game_width + width;
+            gap_top_y = get_random_uniform(game_height - gap_height);
+        }
+
+        Gfx::FloatRect top_rect() const
+        {
+            return { x - width, 0, width, gap_top_y };
+        }
+
+        Gfx::FloatRect bottom_rect() const
+        {
+            return { x - width, gap_top_y + gap_height, width, game_height - gap_top_y - gap_height };
+        }
+    };
+
+    struct Cloud {
+        const Vector<RefPtr<Gfx::Bitmap>> cloud_bitmaps {
+            Gfx::Bitmap::load_from_file("/res/icons/flappybug/cloud_0.png"),
+            Gfx::Bitmap::load_from_file("/res/icons/flappybug/cloud_1.png"),
+            Gfx::Bitmap::load_from_file("/res/icons/flappybug/cloud_2.png"),
+        };
+        float x;
+        float y;
+        int bitmap_id;
+
+        Cloud()
+        {
+            bitmap_id = get_random_uniform(cloud_bitmaps.size());
+            x = get_random_uniform(game_width);
+            y = get_random_uniform(game_height / 2) + bitmap()->height();
+        }
+
+        void reset()
+        {
+            bitmap_id = get_random_uniform(cloud_bitmaps.size());
+            x = game_width + bitmap()->width();
+            y = get_random_uniform(game_height / 2) + bitmap()->height();
+        }
+
+        RefPtr<Gfx::Bitmap> bitmap() const
+        {
+            return cloud_bitmaps[bitmap_id];
+        }
+
+        Gfx::IntRect rect() const
+        {
+            return { (int)x - bitmap()->width(), (int)y - bitmap()->height(), bitmap()->width(), bitmap()->height() };
+        }
+    };
+
+    Bug m_bug;
+    Obstacle m_obstacle;
+    Cloud m_cloud;
+    bool m_active;
+    Optional<float> m_highscore;
+    float m_last_score;
+    float m_difficulty;
+    float m_restart_cooldown;
```

Comment:
All of these primitive types are missing default initialisers.

Reply:
Fixed, thanks!